### ISSUE TITLE
ENABLING HIL MODE IN JSBSIM BRIDGE

### ISSUE THREAD

**Issue by xfahad100** [Auterion/px4-jsbsim-bridge#27]
it seems like it is only sitl enabled. I have successfully enabled serial by changing mavlink tcp port to ttyACM0 in the configs file.
I want to enable HIL mode and serial communication at the same time. a little help would be appreciated. thanks

**Comment by Jaeyoung-Lim** [comment]
@xfahad100 I created a draft pull request from the WIP branch I was working on in #28 

Please feel free to move it forward

**Comment by xfahad100** [comment]
serial connectivity is resolved, but for hitl simulation, connectivity of gcs udp port is also required.

**Comment by xfahad100** [comment]
hil mode is achieved. px4 is sending data through mavlink to qgc in hil simulation

**Comment by xfahad100** [comment]
I will close this issue in 24 hours, thanks for your time.

**Comment by Jaeyoung-Lim** [comment]
@xfahad100 Awesome, so the PR just worked out of the box for you?

**Comment by xfahad100** [comment]
@Jaeyoung-Lim No , I had to make some changes related to connectivity with Qgroundcontrol.

**Comment by xfahad100** [comment]
They were very minor changes . all credit goes to you! @Jaeyoung-Lim

**Comment by Jaeyoung-Lim** [comment]
@xfahad100 Would you mind making a PR with your changes included?

**Comment by xfahad100** [comment]
@Jaeyoung-Lim Yes I will in few days! I just need to comment outunnecessary changes.

**Comment by Jaeyoung-Lim** [comment]
@xfahad100 Awesome, thanks! I will keep this open until it is fully resolved and merged :smile:

**Comment by xfahad100** [comment]
#35 #36 merge them @Jaeyoung-Lim

**Comment by Jaeyoung-Lim** [comment]
@xfahad100 Is there a reason those two are separated in two PRs?

**Comment by xfahad100** [comment]
@Jaeyoung-Lim uhh no,, I'm a newbie on github and I cant figure out  how to compare changes that I've made in 2 files. So there are 2 prs for each file ..... sorry for the mess

**Comment by Jaeyoung-Lim** [comment]
@xfahad100 I tried your PR, but was not able to get GPS position. Were you able to fly with the changes you made?

**Comment by xfahad100** [comment]
ACM0 is workin fine but USB0(telemetry) is giving tx overflow error.

**Comment by Jaeyoung-Lim** [comment]
> ACM0 is workin fine

Can you confirm that you can fly with the vehicle in the simulation? which airframe were you using?

**Comment by Jaeyoung-Lim** [comment]
@xfahad100 Any updates?

**Comment by xfahad100** [comment]
It starts flying by itself in HIL mode and does not respond to px4 data....

**Comment by Jaeyoung-Lim** [comment]
@xfahad100 Are you saying that it didn't work for you?

**Comment by xfahad100** [comment]
@Jaeyoung-Lim  is mag sensor plugin not working at the moment?
I will share my results once I am able to fly my vehicle.

**Comment by Jaeyoung-Lim** [comment]
@xfahad100 Now I am a bit confused. You are saying it worked before and not anymore?

**Comment by xfahad100** [comment]
@Jaeyoung-Lim bruhh I never said that.. I am now able to get gps location but I am not able to fly ..

**Comment by Jaeyoung-Lim** [comment]
@xfahad100 Yes exactly. I have asked you multiple times if it was working, but you never answered. So are you saying it has never worked?

**Comment by xfahad100** [comment]
I'm working on it n its almost done...

**Comment by Jaeyoung-Lim** [comment]
@xfahad100 I am assuming that means it never worked for you.

**Comment by xfahad100** [comment]
@Jaeyoung-Lim it is working for me right now

**Comment by xfahad100** [comment]
At least it takes input from rc and throw it out to jsbsim

**Comment by xfahad100** [comment]
Bro I am doing hitl successfully with px4 and jsbsim and also implementing autopilot

**Comment by xfahad100** [comment]
But I have excluded qgroundcontrol
